packages/plugin-core/src/utils/md.ts

```diff
@@ -313,18 +325,19 @@ export const getReferenceAtPosition = (
       }
     }
     // if not, it could be a frontmatter tag
-    let parsed: ReturnType<typeof parseFrontmatter> | undefined;
-    const noteAST = MDUtilsV5.procRemarkParse(
-      { mode: ProcMode.NO_DATA },
-      {}
-    ).parse(document.getText());
-    visit(noteAST, [DendronASTTypes.FRONTMATTER], (frontmatter: YAML) => {
-      parsed = parseFrontmatter(frontmatter);
-      return false; // stop traversing, there is only one frontmatter
-    });
-    if (parsed) {
-      const tags = getFrontmatterTags(parsed);
-      for (const tag of tags) {
+    // only parse if this is a dendron note
+    if (
+      !(await WorkspaceUtils.isDendronNote({
+        wsRoot,
+        vaults,
+        fpath: document.uri.fsPath,
+      }))
+    ) {
+      return null;
+    }
+    const maybeTags = RemarkUtils.extractFMTags(document.getText());
```

Comment:
No need to add block this PR for it, but we could also limit how much of the text we grab here, like `document.getText(new vscode.Range(0, 0, 300, 0))`. This could cause confusion if someone does legitimately write a note that has more than 300 lines of frontmatter, but it would also help in actual notes that are very long.

Reply:
in terms of parsing text, vscode and a lot of editors have a hard limit of what they're willing to parse (eg. don't pass if file is larger than 1MB). we should do something similar and have a metric for how many times we hit that. will add as a next item